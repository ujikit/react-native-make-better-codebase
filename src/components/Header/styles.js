//package import
import { StyleSheet } from 'react-native';

//local import
import { COLORS } from '../../configs';

export default StyleSheet.create({
  wrapHeader: {
    // paddingTop: Platform.OS === 'android' ? StatusBar.currentHeight : 0,
    zIndex: 1,
    backgroundColor: '#fff',
  },
  shadowHeader: {
    elevation: 2,
    shadowColor: COLORS.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.15,
    shadowRadius: 1.41,
  },
  containerHeader: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 16,
  },
});
