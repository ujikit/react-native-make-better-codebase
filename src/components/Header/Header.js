//package import
import React, { memo } from 'react';
import { SafeAreaView, View } from 'react-native';
import PropTypes from 'prop-types';

//local import
import styles from './styles';
import { COLORS } from '../../configs';

const Header = ({ children, color, shadow, styleWrap, styleContainer }) => {
  return (
    <SafeAreaView
      style={[
        styles.wrapHeader,
        styleWrap,
        shadow && styles.shadowHeader,
        { backgroundColor: color },
      ]}
    >
      <View style={[styles.containerHeader, styleContainer]}>{children}</View>
    </SafeAreaView>
  );
};

Header.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  shadow: PropTypes.bool,
  styleWrap: PropTypes.oneOfType([PropTypes.object, PropTypes.any]),
  styleContainer: PropTypes.oneOfType([PropTypes.object, PropTypes.any]),
};

Header.defaultProps = {
  children: <View />,
  color: COLORS.primaryWhite,
  shadow: true,
  styleWrap: {},
  styleContainer: {},
};

export default memo(Header);
