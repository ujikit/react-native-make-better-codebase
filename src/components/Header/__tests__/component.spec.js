import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '..';

describe('Some component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
