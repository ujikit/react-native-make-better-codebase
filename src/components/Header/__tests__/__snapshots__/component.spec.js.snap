// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "zIndex": 1,
      },
      Object {},
      Object {
        "elevation": 2,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 1.41,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingVertical": 16,
        },
        Object {},
      ]
    }
  >
    <View />
  </View>
</RCTSafeAreaView>
`;
